src/parser/ir/ir_parser.h

```diff
@@ -50,8 +63,9 @@ class IrProgramParser {
   // Operation(nullptr, Operator(core.minus), ..., nullptr )}
   absl::flat_hash_map<std::string, Value> value_map_;
   absl::flat_hash_map<std::string, const Block*> block_map_;
-  IRContext context_;
-  Module module_;
+  std::unique_ptr<IRContext> context_;
+  std::unique_ptr<SsaNames> ssa_names_;
+  std::unique_ptr<Module> module_;
```

Comment:
Remove these fields, as mentioned above.

Reply:
As Gogul pointed out, this suggestion may not make sense, but consider making one `std::unique_ptr` to a `Result`.

Reply:
See my comments above on why `std::unique_ptr` on result did not work